### ISSUE TITLE
CI errors on master

### ISSUE THREAD

**Issue by rok-cesnovar** [stan-dev/stanc3#495]
We keep getting errors on Jenkins tests like these https://jenkins.mc-stan.org/blue/organizations/jenkins/stanc3/detail/master/620/pipeline

This is the error:
```
'example-models/regression_tests/matrix_indexing.stan' had fails '[]' and errors '['Did not compile!']'
'example-models/bugs_examples/vol2/hearts/hearts.stan' had fails '[]' and errors '['Did not compile!']'
script returned exit code 255
```
But I tried compiling these models locally and everything is fine on master and STANC2. Any ideas?

**Comment by seantalts** [comment]
Ah, looks like the `pow` thing didn't work. If you search the build log of the failing step for the model name you will see the compile output for it. ../example-models/regression_tests/matrix_indexing.hpp:425:24: error: call to 'pow' is ambiguous

**Comment by rok-cesnovar** [comment]
Ah, we probably need to wait for this then: https://github.com/stan-dev/math/pull/1810

Thanks for taking a look!

**Comment by bbbales2** [comment]
This looks more like std::pow isn't getting included in the namespace.

The error is:

```
../example-models/bugs_examples/vol1/rats/rats_vec.hpp
../example-models/regression_tests/matrix_indexing.hpp:425:24: error: call to 'pow' is ambiguous
                  ((((-pow(
                       ^~~
../example-models/regression_tests/matrix_indexing.hpp:762:14: note: in instantiation of function template specialization 'matrix_indexing_model_namespace::matrix_indexing_model::log_prob<false, false, double>' requested here
      return log_prob<propto__,jacobian__,T_>(vec_params_r, vec_params_i, pstream);
             ^
/mnt/nvme1n1/workspace/stanc3_master/performance-tests-cmdstan/cmdstan/stan/src/stan/model/model_base_crtp.hpp:90:50: note: in instantiation of function template specialization 'matrix_indexing_model_namespace::matrix_indexing_model::log_prob<false, false, double>' requested here
    return static_cast<const M*>(this)->template log_prob<false, false, double>(
                                                 ^
../example-models/regression_tests/matrix_indexing.hpp:198:3: note: in instantiation of member function 'stan::model::model_base_crtp<matrix_indexing_model_namespace::matrix_indexing_model>::log_prob' requested here
  ~matrix_indexing_model() { }
  ^
/usr/include/x86_64-linux-gnu/bits/mathcalls.h:153:17: note: candidate function
__MATHCALL_VEC (pow,, (_Mdouble_ __x, _Mdouble_ __y));
                ^
/mnt/nvme1n1/workspace/stanc3_master/performance-tests-cmdstan/cmdstan/stan/lib/stan_math/stan/math/rev/fun/pow.hpp:320:26: note: candidate function
inline std::complex<var> pow(const std::complex<var>& x, int y) {
                         ^
/mnt/nvme1n1/workspace/stanc3_master/performance-tests-cmdstan/cmdstan/stan/lib/stan_math/stan/math/rev/fun/pow.hpp:140:12: note: candidate function [with T = int, $1 = void]
inline var pow(const var& base, T exponent) {
           ^
/mnt/nvme1n1/workspace/stanc3_master/performance-tests-cmdstan/cmdstan/stan/lib/stan_math/stan/math/rev/fun/pow.hpp:180:12: note: candidate function [with T = double, $1 = void]
```

Which is saying that since it can't find a ```pow(double, int)``` version of pow it's trying to cast to other things, but there are multiple options.

#1810 should only affect mingw as well. This is clang.

**Comment by rok-cesnovar** [comment]
Thanks both. I dug deeper and found that this problem occurs on compiling with the nightly build of stanc3 (previous master - without std::pow). I guess this is used for comparison?
And with the complex stuff in Math the old nightly build wont compile - hence the error.
But this kind of puts us in a deadlock: old nightly build is failing and a new nightly build cant be built because the old one is failing.

I went in and manually updated (replaced) the nightly build with binaries that were built with the test-binaries job on master (with std:pow fix).
Why the std::pow PR passed remains a mystery to me. Closing as order was restored with an automatically built nightly build.
